### PR TITLE
create a toml package

### PR DESCRIPTION
**Pull request by rybit**

### CODE REVIEW

File: ntoml/netlify_toml.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	"path"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"github.com/pkg/errors"
+)
+
+const DefaultFilename = "netlify.toml"
+
+type NetlifyToml struct {
```
Comment (calavera) [inline ntoml/netlify_toml.go]: This doesn't match the configuration in buildbot. Let's make sure they are compatible:

https://github.com/netlify/buildbot/blob/master/bot/configuration/configuration.go#L19-L31

Comment (rybit) [inline ntoml/netlify_toml.go]: should be addressed

**Comment by calavera** [comment]
LGTM

**Comment by brycekahle** [comment]
What is the benefit of having the configuration definitions here instead of in the projects that use them?

**Comment by rybit** [comment]
it is because we don't have a place to specify "what can you set" in the toml file. It leads to overlapping configuration and "I think you can do that...". It also doesn't let us define some common structures (e.g. settings).

File: ntoml/netlify_toml.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+func LoadFrom(paths ...string) (*NetlifyToml, error) {
+	if len(paths) == 0 {
+		return nil, errors.New("No paths specified")
+	}
+
+	out := new(NetlifyToml)
+
+	for _, p := range paths {
+		if _, err := os.Stat(p); err == nil {
```
[the remark this answers is not in the document]
Comment (brycekahle) [reply ntoml/netlify_toml.go]: The reason for this is that anything can happen on the filesystem between the `Stat` and `ReadFile` calls. If you want to ignore paths that don't exist, best to do it on `ReadFile`.

File: ntoml/netlify_toml.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+			data, ferr := ioutil.ReadFile(p)
+			if ferr != nil {
+				return nil, errors.Wrapf(ferr, "Error while reading in file %s.", p)
+			}
+
+			if _, derr := toml.Decode(string(data), out); derr != nil {
+				return nil, errors.Wrapf(err, "Error while decoding file %s", p)
```
Comment (brycekahle) [inline ntoml/netlify_toml.go]: should be `derr` instead of `err`.

File: ntoml/netlify_toml.go [hunk 4]
```diff
@@ -71,14 +71,14 @@ func LoadFrom(paths ...string) (*NetlifyToml, error) {
...
-			data, ferr := ioutil.ReadFile(p)
-			if ferr != nil {
-				return nil, errors.Wrapf(ferr, "Error while reading in file %s.", p)
-			}
+		data, ferr := ioutil.ReadFile(p)
+		if ferr != nil {
+			return nil, errors.Wrapf(ferr, "Error while reading in file %s.", p)
+		}
 
+		if !os.IsNotExist(ferr) {
```
Comment (brycekahle) [inline ntoml/netlify_toml.go]: `ferr` will always be `nil` here. You need to check `os.IsNotExist` first.

**Pull request merged**